contracts/src/system/PublicCallbacks.sol

```diff
@@ -113,15 +114,15 @@ contract PublicCallbacks is Initializable {
             popCurrentCallback();
         }
 
-        internalRefund(gasRefundValue, target);
+        internalRefund(gasRefundValue, target, callbackId);
         payForCallback(paymentToCoinbase);
     }
 
-    function internalRefund(uint256 gasRefund, address to) internal {
+    function internalRefund(uint256 gasRefund, address to, uint256 callbackId) internal {
         // 22k is the max refund gas limit; 21k for a call and a bit for any accounting the contract might have.
         // ordinarily such accounting should be prepared for beforehand in the callback they pay for, but we give them a
         // slight buffer. 
-        (bool success, ) = to.call{value: gasRefund, gas: 22000}(""); 
+        (bool success, ) = to.call{value: gasRefund, gas: 35000}(abi.encodeWithSignature("handleRefund(uint256)", callbackId)); 
```

Comment:
I don't understand what's going on here.
Is there a method somewhere: `handleRefund`?

Reply:
cool. pls add this as comment